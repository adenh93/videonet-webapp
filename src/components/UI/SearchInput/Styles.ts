import styled from "styled-components";
import { deviceWidths } from "../../../CONSTANTS";

export const InputContainer = styled.span`
  position: relative;
`;

export const InputIcon = styled.img`
  width: 2.6rem;
  position: absolute;
  z-index: 3;
  transform: translate(130%, 65%);
`;

export const SearchButton = styled.button`
  position: absolute;
  cursor: pointer;
  background-color: ${({ theme }) => theme.palette.button.primary};
  color: ${({ theme }) => theme.palette.text.main};
  font-size: ${({ theme }) => theme.fontSize.button};
  font-weight: ${({ theme }) => theme.fontWeights.semiBold};
  font-family: "Lato", sans-serif;
  letter-spacing: 0.03em;
  height: 6rem;
  width: 12.5rem;
  transform: translateX(calc(-100% + 1px));
  border: none;
  border-top-right-radius: ${({ theme }) => theme.borderRadius.rounded};
  border-bottom-right-radius: ${({ theme }) => theme.borderRadius.rounded};

  @media (max-width: ${deviceWidths.tablet}) {
    position: static;
    margin-left: 100%;
    margin-top: 1rem;
    border-radius: ${({ theme }) => theme.borderRadius.rounded};
    filter: drop-shadow(${({ theme }) => theme.shadows.dark});
    width: 100%;
  }
`;
