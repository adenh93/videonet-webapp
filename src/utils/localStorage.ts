import { FilmDetails, SearchResult } from "../graphql/types";

type LocalStorageCollection = "WATCHED_ITEMS" | "LIKED_ITEMS";

export const setItem = <T>(type: LocalStorageCollection, item: T): void =>
  localStorage.setItem(type, JSON.stringify(item));

export const getItems = <R>(type: LocalStorageCollection): R[] => {
  const items = localStorage.getItem(type);
  if (!items) return [];
  return JSON.parse(items) as R[];
};

export const getWatchList = (): SearchResult[] =>
  getItems<SearchResult>("WATCHED_ITEMS");

export const isInWatchList = (movieId: number): boolean =>
  getWatchList().find((m) => m.id === movieId) !== undefined;

export const addToWatchList = (movie: FilmDetails): void => {
  const movies = getWatchList();
  if (!movies.find((m) => m.id === movie.id))
    setItem("WATCHED_ITEMS", [...movies, movie]);
};

export const removeFromWatchList = (movieId: number): void =>
  setItem(
    "WATCHED_ITEMS",
    getWatchList().filter((m) => m.id !== movieId)
  );

export const getLikedMovies = (): number[] => getItems<number>("LIKED_ITEMS");

export const isLiked = (movieId: number): boolean =>
  getLikedMovies().find((item) => item === movieId) !== undefined;

export const likeMovie = (movieId: number): void => {
  const likedItems = getLikedMovies();
  if (!likedItems.find((item) => item === movieId))
    setItem("LIKED_ITEMS", [...likedItems, movieId]);
};

export const unlikeMovie = (movieId: number): void => {
  setItem(
    "LIKED_ITEMS",
    getLikedMovies().filter((item) => item !== movieId)
  );
};

export const toggleWatched = (movie: FilmDetails): boolean => {
  const watched = isInWatchList(movie.id);
  watched ? removeFromWatchList(movie.id) : addToWatchList(movie);
  return !watched;
};

export const toggleLiked = (movieId: number): boolean => {
  const liked = isLiked(movieId);
  liked ? unlikeMovie(movieId) : likeMovie(movieId);
  return !liked;
};
