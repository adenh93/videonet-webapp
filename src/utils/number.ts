import numeral from "numeral";

export const formatNumber = (n: Number): String => numeral(n).format("0,0");
