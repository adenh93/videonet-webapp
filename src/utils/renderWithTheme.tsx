import React, { ReactElement } from "react";
import renderer from "react-test-renderer";
import { render } from "@testing-library/react";
import { ThemeProvider } from "styled-components";
import { BrowserRouter as Router } from "react-router-dom";
import theme from "../theme";

export const renderWithTheme = (component: ReactElement) =>
  renderer.create(
    <ThemeProvider theme={theme}>
      <Router>{component}</Router>
    </ThemeProvider>
  );

export const rtlWithTheme = (component: ReactElement) =>
  render(
    <ThemeProvider theme={theme}>
      <Router>{component}</Router>
    </ThemeProvider>
  );
