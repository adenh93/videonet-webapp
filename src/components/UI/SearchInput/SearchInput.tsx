import React, { forwardRef } from "react";
import Input from "../Input";
import { search } from "../../../assets/icons";
import { InputContainer, InputIcon, SearchButton } from "./Styles";

interface SearchInputProps extends React.InputHTMLAttributes<HTMLInputElement> {
  onSearch: (e: React.FormEvent) => void;
}

const SearchInput = forwardRef<HTMLInputElement, SearchInputProps>(
  ({ onSearch, ...props }, ref) => (
    <form onSubmit={onSearch}>
      <InputContainer>
        <InputIcon src={search} />
        <Input {...props} ref={ref} />
        <SearchButton type="submit">Search</SearchButton>
      </InputContainer>
    </form>
  )
);

export default SearchInput;
