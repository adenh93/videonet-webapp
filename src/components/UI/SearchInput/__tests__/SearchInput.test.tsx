import React from "react";
import { renderWithTheme } from "../../../../utils/renderWithTheme";
import SearchInput from "../SearchInput";
import "jest-styled-components";

describe("SearchInput UI component", () => {
  it("renders correctly", () => {
    const tree = renderWithTheme(
      <SearchInput onSearch={() => {}}>Test</SearchInput>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
