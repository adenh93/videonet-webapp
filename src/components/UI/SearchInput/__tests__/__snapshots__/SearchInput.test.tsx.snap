// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchInput UI component renders correctly 1`] = `
.c2 {
  color: #333333;
  font-size: 1.8rem;
  font-weight: 300;
  background-color: #f9f9f9;
  font-family: "Lato",sans-serif;
  -webkit-letter-spacing: 0.03em;
  -moz-letter-spacing: 0.03em;
  -ms-letter-spacing: 0.03em;
  letter-spacing: 0.03em;
  height: 6rem;
  width: 73.5rem;
  border-radius: 20rem;
  padding: 1rem 14rem 1rem 8rem;
  border: 0;
  -webkit-filter: drop-shadow(0 0.4rem 0.4rem rgba(0,0,0,0.25));
  filter: drop-shadow(0 0.4rem 0.4rem rgba(0,0,0,0.25));
}

.c0 {
  position: relative;
}

.c1 {
  width: 2.6rem;
  position: absolute;
  z-index: 3;
  -webkit-transform: translate(130%,65%);
  -ms-transform: translate(130%,65%);
  transform: translate(130%,65%);
}

.c3 {
  position: absolute;
  cursor: pointer;
  background-color: #BA68C8;
  color: #f9f9f9;
  font-size: 2rem;
  font-weight: 600;
  font-family: "Lato",sans-serif;
  -webkit-letter-spacing: 0.03em;
  -moz-letter-spacing: 0.03em;
  -ms-letter-spacing: 0.03em;
  letter-spacing: 0.03em;
  height: 6rem;
  width: 12.5rem;
  -webkit-transform: translateX(calc(-100% + 1px));
  -ms-transform: translateX(calc(-100% + 1px));
  transform: translateX(calc(-100% + 1px));
  border: none;
  border-top-right-radius: 20rem;
  border-bottom-right-radius: 20rem;
}

<form
  onSubmit={[Function]}
>
  <span
    className="c0"
  >
    <img
      className="c1"
      src="search.svg"
    />
    <input
      className="c2"
    >
      Test
    </input>
    <button
      className="c3"
      type="submit"
    >
      Search
    </button>
  </span>
</form>
`;
